integrations/parsely.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Integration: Parse.ly.
+ *
+ * @package Automattic\VIP\Integrations
+ */
+
+namespace Automattic\VIP\Integrations;
+
+/**
+ * Loads Parse.ly VIP Integration.
+ *
+ * @private
+ */
+class ParselyIntegration extends Integration {
+	/**
+	 * Loads the plugin.
+	 *
+	 * @private
+	 */
+	public function load(): void {
+		// Do not load plugin if already loaded by customer code.
+		if ( class_exists( 'Parsely\Parsely' ) ) {
+			return;
+		}
+
+		// Activating the integration via setting constant whose implementation is already
+		// handled via Automattic\VIP\WP_Parsely_Integration (ideally we should move
+		// all the implementation here such that there will be only one way of managing
+		// the plugin).
+		define( 'VIP_PARSELY_ENABLED', true );
+
+		add_filter( 'wp_parsely_credentials', array( $this, 'wp_parsely_credentials_callback' ) );
+	}
+
+	/**
+	 * Callback for `wp_parsely_credentials` filter.
+	 */
+	protected function wp_parsely_credentials_callback(): array {
+		return $this->get_config();
```

Comment:
I  was thinking something like this:

```suggestion
	protected function wp_parsely_credentials_callback( $original_credentials ): array {
		$config = $this->get_config();
		
		if ( empty( $config ) ) {
			return $original_credentials;
		}

		$new_credentials = array(
			'is_managed' => true,
			'site_id' => $config['site_id'] ?? null,
			'api_secret' => $config['api_secret'] ?? null,
		};

		return $new_credentials;
```

Reply:
This looks easy to maintain (thanks for the suggestion), will incorporate this before merging but without `$original_credentials` param because `wp_parsely_credentials` filter on plugin doesn't pass any parameter.

Reply:
Interesting. Filters should always be passing something (even if it's a null value). I think we should assume that's the case, and update the filter in the plugin to do so as well.

Reply:
I think I have mistaken the filter is passing `false`. Okay will update it as suggested.

Reply:
Added in https://github.com/Automattic/vip-go-mu-plugins/pull/4660/commits/9839b7907cc6cfd36ecb57ebcfb1ff40bf5ea928, will add `is_managed` key later (have a separate ticket for it)